test/joseph/JosephTreasury.t.sol

```diff
@@ -0,0 +1,245 @@
+        prepareIpToken(_ipTokenDai, address(mockCase0JosephDai));
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(_daiMockedToken), TestConstants.PERCENTAGE_3_18DEC, block.timestamp);
+        vm.prank(_liquidityProvider);
+        mockCase0JosephDai.itfProvideLiquidity(TestConstants.USD_28_000_18DEC, block.timestamp);
+        vm.prank(_userTwo);
+        mockCase0MiltonDai.itfOpenSwapPayFixed(
+            block.timestamp,
+            TestConstants.USD_10_000_18DEC,
+            TestConstants.PERCENTAGE_6_18DEC,
+            TestConstants.LEVERAGE_18DEC
+        );
+        vm.startPrank(_admin);
+        mockCase0JosephDai.setCharlieTreasuryManager(_userThree);
+        mockCase0JosephDai.setCharlieTreasury(_userOne);
+        vm.stopPrank();
+        // when
+        vm.prank(_userThree);
+        mockCase0JosephDai.transferToCharlieTreasury(transferredAmount);
+        // then
+        MiltonStorageTypes.ExtendedBalancesMemory memory balance = miltonStorageDai.getExtendedBalance();
+        uint256 actualERC20BalanceCharlieTreasury = _daiMockedToken.balanceOf(_userOne);
+        uint256 actualERC20BalanceMilton = _daiMockedToken.balanceOf(address(mockCase0MiltonDai));
+        uint256 actualPublicationFeeBalanceMilton = balance.iporPublicationFee;
+        assertEq(actualERC20BalanceCharlieTreasury, expectedERC20BalanceCharlieTreasury);
+        assertEq(actualERC20BalanceMilton, expectedERC20BalanceMilton);
+        assertEq(actualPublicationFeeBalanceMilton, expectedPublicationFeeBalanceMilton);
+    }
+
+    function testShouldNotTransferToTreasuryWhenCallerIsNotTreasuryTransferer() public {
+        // given
+        ItfIporOracle iporOracle =
+            getIporOracleAsset(_userOne, address(_daiMockedToken), TestConstants.TC_DEFAULT_EMA_18DEC_64UINT);
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(_daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        MockCase0MiltonDai mockCase0MiltonDai = getMockCase0MiltonDai(
+            address(_daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        MockCase0JosephDai mockCase0JosephDai = getMockCase0JosephDai(
+            address(_daiMockedToken),
+            address(_ipTokenDai),
+            address(mockCase0MiltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, _daiMockedToken, address(mockCase0JosephDai), address(mockCase0MiltonDai));
+        prepareMilton(mockCase0MiltonDai, address(mockCase0JosephDai), address(stanleyDai));
+        prepareJoseph(mockCase0JosephDai);
+        prepareIpToken(_ipTokenDai, address(mockCase0JosephDai));
+        // when
+        vm.expectRevert("IPOR_404");
+        vm.prank(_userThree);
+        mockCase0JosephDai.transferToTreasury(100);
+    }
+
```

Comment:
Missing test ""should NOT transfer Publication Fee to Charlie Treasury - Treasury Manager address incorrect""